kentik_api_library/kentik_api/synthetics/api_transport_grpc.py

```diff
@@ -0,0 +1,880 @@
+        response = self._admin.PatchAgent(request=patch_agent_req, metadata=self._credentials, target=self._url)
+        return pb_to_agent(response.agent)
+
+    def agent_delete(self, **kwargs) -> None:
+        id = str(kwargs["id"])
+        self._admin.DeleteAgent(DeleteAgentRequest(id=id), metadata=self._credentials, target=self._url)
+
+    def get_health_for_tests(self, **kwargs) -> List[TestHealth]:
+        ids = [str(id) for id in kwargs["test_ids"]]
+        agents = [str(id) for id in kwargs["agent_ids"]]
+        tasks = [str(id) for id in kwargs["task_ids"]]
+        start = Timestamp(seconds=int(kwargs["start_time"].timestamp()))
+        end = Timestamp(seconds=int(kwargs["end_time"].timestamp()))
+        get_health_req = GetHealthForTestsRequest(
+            ids=ids,
+            start_time=start,
+            end_time=end,
+            agent_ids=agents,
+            task_ids=tasks,
+            augment=kwargs["augment"],
+        )
+        response = self._data.GetHealthForTests(request=get_health_req, metadata=self._credentials, target=self._url)
+        return pb_to_health_tests(response.health)
+
+    def get_trace_for_test(self, **kwargs) -> GetTraceForTestResponse:
+        id = str(kwargs["id"])
+        agents = [str(id) for id in kwargs["agent_ids"]]
+        ips = [str(ip) for ip in kwargs["target_ips"]]
+        start = Timestamp(seconds=int(kwargs["start_time"].timestamp()))
+        end = Timestamp(seconds=int(kwargs["end_time"].timestamp()))
+        get_trace_req = GetTraceForTestRequest(
+            id=id,
+            start_time=start,
+            end_time=end,
+            agent_ids=agents,
+            target_ips=ips,
+        )
+        response = self._data.GetTraceForTest(request=get_trace_req, metadata=self._credentials, target=self._url)
+        return pb_to_trace_response(response)
+
+
+def populate_test_from_pb(v: pbTest, out: SynTest) -> None:
+    out.name = v.name
+    out.type = TestType(v.type)
+    out.status = PB_TO_TEST_STATUS[v.status]
+    out.deviceId = ID(v.device_id)
+    out._id = ID(v.id)
+    out._cdate = pb_to_datetime_iso(v.cdate)
+    out._edate = pb_to_datetime_iso(v.edate)
+
+    settings = SynTestSettings()
+    pupulate_settings_from_pb(v.settings, settings)
+    out.settings = settings
+
+
+# disable no-member linting as gRPC structures are wrongly recognized as to have missing attributes
+# pylint: disable=no-member
+def pb_from_test(v: SynTest) -> pbTest:
+    out = pbTest()
+    out.name = v.name
+    out.device_id = str(v.deviceId)
+    out.id = str(v.id)
+    out.type = v.type.value
+    out.status = reverse_map(PB_TO_TEST_STATUS, v.status)
+    out.settings.CopyFrom(pb_from_settings(v.settings))
+    return out
+
+
+def pupulate_ping_trace_test_from_pb(v: pbTest, out: PingTraceTest) -> None:
+    populate_test_from_pb(v, out)
+    populate_ping_test_settings_from_pb(v.settings, out.settings)
+
+
+def populate_mesh_test_from_pb(v: pbTest, out: MeshTest) -> None:
```

Comment:
I wonder why is the `pbTest` argument named `v`. I like short name, but `t` seems much more fitting here.

Reply:
v for src "value"

Reply:
changed v to src, and used dst for the other way around conversion methods